typed/typed.go

```diff
@@ -109,12 +109,28 @@ func (tv TypedValue) Merge(pso *TypedValue) (*TypedValue, error) {
 // match), or an error will be returned. Validation errors will be returned if
 // the objects don't conform to the schema.
 func (tv TypedValue) Compare(rhs *TypedValue) (c *Comparison, err error) {
+	return tv.CompareWithout(rhs, nil)
+}
+
+// CompareWithout compares the two objects, but ignores the provided set of fields.
+// See the comments on the `Comparison` struct for details on the return value.
+//
+// tv and rhs must both be of the same type (their Schema and TypeRef must
+// match), or an error will be returned. Validation errors will be returned if
+// the objects don't conform to the schema.
+func (tv TypedValue) CompareWithout(rhs *TypedValue, ignored *fieldpath.Set) (c *Comparison, err error) {
 	c = &Comparison{
 		Removed:  fieldpath.NewSet(),
 		Modified: fieldpath.NewSet(),
 		Added:    fieldpath.NewSet(),
 	}
+	if ignored == nil {
+		ignored = fieldpath.NewSet()
+	}
 	_, err = merge(&tv, rhs, func(w *mergingWalker) {
+		if ignored.HasPartial(w.path) {
```

Comment:
I didn't really expect the merge code to be altered by the ignored field. Can't we just remove the fields from the set after we're done?

Reply:
We could, but that would mean to go over all managers. I thought this way is more efficient as we could do the other thing in k/k as well.

Reply:
not if you do it at the right time? We can change the list of fields changed before we update the other managers (I think).

Reply:
You mean removing the fields from *typed.Comparison?
That would work, but needs to go over added, modified and removed again and then do the same logic/remove from there.

Reply:
Yeah, I think that's better than altering the merge code though